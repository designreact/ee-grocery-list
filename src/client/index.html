<!DOCTYPE html>
<html>
<head>
  <meta charset='utf-8'>
  <meta http-equiv='X-UA-Compatible' content='IE=edge'>
  <title>React Shopping List</title>
  <meta name='viewport' content='width=device-width, initial-scale=1'>
  <link rel='stylesheet' type='text/css' media='screen' href='main.css'>
  <link rel="icon" type="image/x-icon" href="static/favicon.ico">
</head>
<body>
  <div id="root"></div>
</body>
</html>