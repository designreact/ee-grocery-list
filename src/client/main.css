html {
  background-color: #333;
  color: #fff;
  font-family: 'Courier New', Courier, monospace;
  font-size: 16px;
}

.app {
  max-width: 40em;
  padding: 2em;
  margin: 0 auto;
  background-color: #222;
}

.app > h1 {
  margin-top: 0;
}

.list {
  padding: 0;
  list-style: none;
}

.item,
.form-entry {
  display: flex;
  flex-direction: row;
  align-content: flex-end;
}

.item.complete {
  color: #888;
  text-decoration: line-through;
}

.item.complete .app-button.check {
  display: none;
}

.form-label {
  display: block;
  padding: 1em 0;
  border-top: 1px #fff solid;
  font-weight: 600;
}

.item-text,
.form-input {
  flex: 1 0 0;
}

.form-input {
  border-radius: .2em;
}

.app-button {
  width: 2em;
  height: 2em;
  padding: .5em;
  border: none;
  margin-left: 1em;
  border-radius: .2em;
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  background-size: 80%;
}

.app-button.add {
  background-image: url('/static/plus.svg');
  background-color: #008;
}

.app-button.delete {
  background-image: url('/static/cancel.svg');
  background-color: #800;
}

.app-button.check {
  background-image: url('/static/check.svg');
  background-color: #080;
}