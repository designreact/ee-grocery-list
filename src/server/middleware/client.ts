import Koa from 'koa';
import fs from 'fs';
import path from 'path';
import Router from 'koa-router';

const indexPath = path.resolve(__dirname, '../../client/index.html');
const jsPath = path.resolve(__dirname, '../../client/main.js');

const clientRouter = new Router();

clientRouter.get('/', (ctx: Koa.Context): void => {
  const html = fs.readFileSync(indexPath);
  ctx.status = 200;
  ctx.type = 'text/html';
  ctx.body = html;
});

clientRouter.get('/main.js', (ctx: Koa.Context): void => {
  const js = fs.readFileSync(jsPath);
  ctx.status = 200;
  ctx.type = 'application/javascript';
  ctx.body = js;
});

export const clientMiddleware = clientRouter.routes();