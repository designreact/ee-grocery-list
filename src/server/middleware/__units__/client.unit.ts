import Koa from "koa";
import { clientMiddleware } from '../client'

test('The client middleware to set the status and body as expected', () => {
  const ctx = {};
  clientMiddleware(ctx as Koa.Context);
  expect(ctx).toMatchSnapshot();
});