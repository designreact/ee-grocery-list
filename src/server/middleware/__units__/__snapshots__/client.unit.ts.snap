// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The client middleware to set the status and body as expected 1`] = `
Object {
  "matched": Array [
    Layer {
      "methods": Array [
        "HEAD",
        "GET",
      ],
      "name": null,
      "opts": Object {
        "end": true,
        "ignoreCaptures": undefined,
        "name": null,
        "prefix": "",
        "sensitive": false,
        "strict": false,
      },
      "paramNames": Array [
        Object {
          "asterisk": true,
          "delimiter": "/",
          "name": 0,
          "optional": false,
          "partial": false,
          "pattern": ".*",
          "prefix": "",
          "repeat": false,
        },
      ],
      "path": "*",
      "regexp": /\\^\\(\\(\\?:\\.\\*\\)\\)\\(\\?:\\\\/\\(\\?=\\$\\)\\)\\?\\$/i,
      "stack": Array [
        [Function],
      ],
    },
  ],
  "router": Router {
    "methods": Array [
      "HEAD",
      "OPTIONS",
      "GET",
      "PUT",
      "PATCH",
      "POST",
      "DELETE",
    ],
    "opts": Object {},
    "params": Object {},
    "stack": Array [
      Layer {
        "methods": Array [
          "HEAD",
          "GET",
        ],
        "name": null,
        "opts": Object {
          "end": true,
          "ignoreCaptures": undefined,
          "name": null,
          "prefix": "",
          "sensitive": false,
          "strict": false,
        },
        "paramNames": Array [],
        "path": "/",
        "regexp": /\\^\\(\\?:\\\\/\\(\\?=\\$\\)\\)\\?\\$/i,
        "stack": Array [
          [Function],
        ],
      },
      Layer {
        "methods": Array [
          "HEAD",
          "GET",
        ],
        "name": null,
        "opts": Object {
          "end": true,
          "ignoreCaptures": undefined,
          "name": null,
          "prefix": "",
          "sensitive": false,
          "strict": false,
        },
        "paramNames": Array [
          Object {
            "asterisk": true,
            "delimiter": "/",
            "name": 0,
            "optional": false,
            "partial": false,
            "pattern": ".*",
            "prefix": "",
            "repeat": false,
          },
        ],
        "path": "*",
        "regexp": /\\^\\(\\(\\?:\\.\\*\\)\\)\\(\\?:\\\\/\\(\\?=\\$\\)\\)\\?\\$/i,
        "stack": Array [
          [Function],
        ],
      },
    ],
  },
}
`;
